Automattic/vip-design-system
# Wizard color contrast improvements + title steps

## Description

**This PR deprecates support for the Step title being a component. It will now only supports titles being strings.**

Changes to improve Wizard Color contrast and title steps.

### Before
<img width="711" alt="CleanShot 2023-03-10 at 16 02 58@2x" src="https://user-images.githubusercontent.com/668251/224349839-557da470-488f-403f-ac7a-f4d316a60d93.png">

### After
<img width="834" alt="CleanShot 2023-03-10 at 16 03 29@2x" src="https://user-images.githubusercontent.com/668251/224349953-775df647-1474-4242-9708-9d68ab636846.png">

This PR includes
* Changing of icons to use a full circle only and change the color based on the status.
* New "Step X of Y", which is printed in the heading so that it's read/announced together with the heading itself. The text is not in full caps for[ improved readability.](https://www.a11yproject.com/posts/how-to-accessible-heading-structure/#all-caps). (screenshots were made before the change)
* Update of the colors to match the design
* Instead of getting the colors from the generic theme, we use a component-approach where the colors we need are in a single component ([see here](https://github.com/Automattic/vip-design-system/pull/189/files#diff-766d29b6d3e7340553dd59e89c2d8069a4bf232d11034f3fdacfabc50c19f319))
* A11Y: I've added a "Status: [Step completed | Current step | Step not completed]" so that screen readers users will know the status too. 
* `summary` step field support, for recapping the information of a single completed or skipped step
* "Change" button, that can be called to activate a single step
* Responsive improvements (now if you resize the screen the icon stays the same and everything is fluid)


![CleanShot 2023-03-13 at 18 09 25@2x](https://user-images.githubusercontent.com/668251/224775249-8ddb5587-b3c2-4a68-bc8a-85b44a5549ba.png)

Here it is on a small screen
<img width="382" alt="CleanShot 2023-03-13 at 18 11 26@2x" src="https://user-images.githubusercontent.com/668251/224775298-d767c1fc-ce97-4dc6-974c-7b965de321dd.png">

## Checklist


## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Pull down PR.
1. `npm run dev`.
1. Open Storybook.
1. check the wizard component https://deploy-preview-189--vip-design-system-components.netlify.app/?path=/story/wizard--with-title-auto-focus
